python_modules/dagster/dagster/_core/definitions/time_window_partition_mapping.py

```diff
@@ -101,7 +101,9 @@ def round_datetime_to_period(dt: datetime, period: ScheduleType) -> datetime:
     if period == ScheduleType.MONTHLY:
         return dt.replace(day=1, hour=0, minute=0, second=0, microsecond=0)
     elif period == ScheduleType.WEEKLY:
-        return (dt - timedelta(days=dt.weekday())).replace(
+        # isoweekday returns 1 for Monday, 7 for Sunday. Our weekly partitions start from every
```

Comment:
Fixes a bug that I came across where mapping a partition key into its corresponding weekly partition key will result in a weekly partition beginning on Monday.

Our weekly partitions definitions start on Sunday. Updates this mapping to map to a weekly partition starting on Sunday.